packages/react/src/components/StructuredList/StructuredList.js

```diff
@@ -5,321 +5,309 @@
+   */
+  border: deprecate(
+    PropTypes.bool,
+    `\nThe prop \`border\` will be removed in the next major version of Carbon.`
+  ),
+
+  /**
+   * Provide the contents of your StructuredListWrapper
+   */
+  children: PropTypes.node,
+
+  /**
+   * Specify an optional className to be applied to the container node
+   */
+  className: PropTypes.string,
+
+  /**
+   * Specify whether your StructuredListWrapper should have selections
+   */
+  selection: PropTypes.bool,
+};
+
+StructuredListWrapper.defaultProps = {
+  selection: false,
+  ariaLabel: 'Structured list section',
+};
+
+export function StructuredListHead(props) {
+  const { children, className, ...other } = props;
+  const classes = classNames(`${prefix}--structured-list-thead`, className);
+
+  return (
+    <div role="rowgroup" className={classes} {...other}>
+      {children}
+    </div>
+  );
 }
 
-const getInstanceId = setupGetInstanceId();
-
-export class StructuredListInput extends Component {
-  static propTypes = {
-    /**
-     * Specify an optional className to be applied to the input
-     */
-    className: PropTypes.string,
-
-    /**
-     * Specify whether the underlying input should be checked by default
-     */
-    defaultChecked: PropTypes.bool,
-
-    /**
-     * Specify a custom `id` for the input
-     */
-    id: PropTypes.string,
-
-    /**
-     * Provide a `name` for the input
-     */
-    name: PropTypes.string,
-
-    /**
-     * Provide an optional hook that is called each time the input is updated
-     */
-    onChange: PropTypes.func,
-
-    /**
-     * Provide a `title` for the input
-     */
-    title: PropTypes.string,
-
-    /**
-     * Specify the value of the input
-     */
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
-  };
-
-  static defaultProps = {
-    onChange: () => {},
-    value: 'value',
-    title: 'title',
-  };
-
-  constructor(props) {
-    super(props);
-    this.uid = this.props.id || getInstanceId();
-  }
-
-  render() {
-    const { className, value, name, title, ...other } = this.props;
-    const classes = classNames(`${prefix}--structured-list-input`, className);
-    return (
-      <input
-        {...other}
-        type="radio"
-        tabIndex={-1}
-        id={this.uid}
-        className={classes}
-        value={value}
-        name={name}
-        title={title}
-      />
-    );
-  }
+StructuredListHead.propTypes = {
+  /**
+   * Provide the contents of your StructuredListHead
+   */
+  children: PropTypes.node,
+
+  /**
+   * Specify an optional className to be applied to the node
+   */
+  className: PropTypes.string,
+};
+
+export function StructuredListBody(props) {
+  const { children, className, ...other } = props;
+  const classes = classNames(`${prefix}--structured-list-tbody`, className);
+  const { labelRows, setLabelRows } = useState(null);
+  const { rowSelected, setRowSelected } = useState(0);
```

Comment:
this is kinda weird, probably something i'm missing, but `setLabel` and `rowSelected` were tracked in state, but unused?

Reply:
AFAIK `StructuredList` hasn't been touched in quite some time and could definitely have some leftover unused hooks that may need to be refactored

Reply:
Totally the case -- just got rid of em